src/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl.py

```diff
@@ -746,7 +754,13 @@ def __call__(
         add_time_ids = add_time_ids.to(device).repeat(batch_size * num_images_per_prompt, 1)
 
         # 8. Denoising loop
-        num_warmup_steps = len(timesteps) - num_inference_steps * self.scheduler.order
+        num_warmup_steps = max(len(timesteps) - num_inference_steps * self.scheduler.order, 0)
+
+        # 7.1 Apply denoising_end
+        if denoising_end is not None:
+            num_inference_steps = int(round(denoising_end * num_inference_steps))
```

Comment:
this resolved the timestamp bar not completing. however, i really liked that emergent property, as it meant any progress bar capturing the tqdm output would "just magically" have the progress bar continue on logically from one to the next. 😥 more work for that result, but i've been needing to refactor how i handle that, anyway...